src/EditorFeatures/Core/Shared/Extensions/SmartIndentExtensions.cs

```diff
@@ -6,9 +6,9 @@
 
 namespace Microsoft.CodeAnalysis.Editor.Shared.Extensions
 {
-    internal static class SmartIndentExtensions
+    internal static class IndentationResultExtensions
     {
-        public static int GetIndentation(this IndentationResult result, ITextView textView, ITextSnapshotLine lineToBeIndented)
+        public static int GetIndentation(this Indentation.IndentationResult result, ITextView textView, ITextSnapshotLine lineToBeIndented)
         {
             var position = new SnapshotPoint(lineToBeIndented.Snapshot, result.BasePosition);
```

Comment:
can we add some comment what this is trying to do and what each position, snapshot is supposed to be refering? subject buffer, data buffer? in what scenario.

Reply:
what guarantee that result.BasePosition is on the given linetoBeindented.Snapshot? why not the indentation result hold onto snapshotpoint on the line itself?

Reply:
okay. so I see 3 different usages case for this method, one for sync indenter and one for blank indenter and one for smart indenter and all seems give different lined to be indented? yet use same base position? feels like there is some assumption made on how things are used together. and not sure what that is.